### PR TITLE
[ARM] __ARM_ARCH macro definition fix

### PR DESCRIPTION
**Pull request by jwestwood921**
This patch changes how the macro __ARM_ARCH is defined to match its defintion in the ACLE. In ACLE 5.4.1, __ARM_ARCH is defined as equal to the major architecture version for ISAs up to and including v8. From v8.1 onwards, its definition is changed to include minor versions, such that for an architecture vX.Y, __ARM_ARCH = X*100 + Y. Before this patch, LLVM defined __ARM_ARCH using only the major architecture version for all architecture versions. This patch adds functionality to define __ARM_ARCH correctly for architectures greater than or equal to v8.1.

### CODE REVIEW

**Comment by pinskia** [comment]
Note the corresponding GCC bug is at https://gcc.gnu.org/bugzilla/show_bug.cgi?id=99312 (and there was a patch posted but never reviewed; https://gcc.gnu.org/pipermail/gcc-patches/2021-March/566322.html).

File: clang/lib/Basic/Targets/AArch64.cpp [hunk 1]
```diff
@@ -367,8 +367,21 @@ void AArch64TargetInfo::getTargetDefines(const LangOptions &Opts,
 
   // ACLE predefines. Many can only have one possible value on v8 AArch64.
   Builder.defineMacro("__ARM_ACLE", "200");
-  Builder.defineMacro("__ARM_ARCH",
-                      std::to_string(ArchInfo->Version.getMajor()));
+
+  // ACLE 5.4.1 ARM/Thumb instruction set architecture
```
Comment (ostannard) [inline clang/lib/Basic/Targets/AArch64.cpp:371]: Nit: the ACLE doesn't use section numbers any more, and and the ARM/Thumb ISAs only apply to AArch32. I think this line of the comment can just be removed.

File: clang/lib/Basic/Targets/ARM.cpp [hunk 2]
```diff
@@ -736,9 +737,17 @@ void ARMTargetInfo::getTargetDefines(const LangOptions &Opts,
   if (!CPUAttr.empty())
     Builder.defineMacro("__ARM_ARCH_" + CPUAttr + "__");
 
-  // ACLE 6.4.1 ARM/Thumb instruction set architecture
-  // __ARM_ARCH is defined as an integer value indicating the current ARM ISA
-  Builder.defineMacro("__ARM_ARCH", Twine(ArchVersion));
+  // ACLE 5.4.1 ARM/Thumb instruction set architecture
```
Comment (ostannard) [inline clang/lib/Basic/Targets/ARM.cpp:740]: This macro doesn't actually have anything to do with the ARM/Thumb ISAs, this line of the comment can just be deleted.

**Review by ostannard — APPROVED** [review]
LGTM, just some minor changes to the comments which don't need re-review.

**Pull request merged**